### ISSUE TITLE
firebase-crashlytics-gradle:2.9.5 not working

### ISSUE THREAD

**Issue by ber4444** [firebase/firebase-android-sdk#4912]
### [READ] Step 1: Are you in the right place?

Yes

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Giraffe
  * Firebase Component: Crashlytics plugin
  * Component version: 2.9.5

#### Steps to reproduce:

Build fails, current workaround is to downgrade to as far back as 2.9.2 due to separate bugs in previous versions

#### Relevant Code:

```
 % ./gradlew clean testProdReleaseUnitTest
...
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:mapProdReleaseSourceSetPaths'.
> Could not determine the dependencies of null.
   > Could not create task ':app:injectCrashlyticsBuildIdsProdRelease'.
      > Cannot invoke method map() on null object
```
usage of the plugin:
```
import com.google.firebase.crashlytics.buildtools.gradle.CrashlyticsExtension
plugins {
    id("com.google.firebase.crashlytics")
}
android {
    buildTypes {
        release {
            the<CrashlyticsExtension>().nativeSymbolUploadEnabled = true
            the<CrashlyticsExtension>().unstrippedNativeLibsDir = file("$rootDir/app/build/intermediates/merged_native_libs/debug/out/lib/armeabi-v7a/")
        }
    }
....
```

**Comment by vlohachov** [comment]
Same issue

**Comment by argzdev** [comment]
Hi @ber4444, thanks for reporting. While our engineers are taking a look into this. I tried creating a new project with NDK setup with the Firebase Crashlytics plugin version `2.9.5`, and I've added the relevant code you've provided, however I'm unable to reproduce the same exception when running the gradle command of:
- `./gradlew mapReleaseSourceSetPaths`
- `./gradlew clean testReleaseUnitTest`

I've tried testing these commands on both Android Studio version:
- `Android Studio Giraffe | 2022.3.1 Beta 1`
- `Android Studio Flamingo | 2022.2.1`

I've also tried testing this on AGP & Gradle plugin versions:
- `8.0.0` & `8.0`
- `7.4.0` & `7.5`

Not sure what I'm missing here. Could you help me reproduce this issue?

**Comment by ber4444** [comment]
We are using 8.1 alpha, please try with that

**Comment by runmael** [comment]
Same problem with the 2.9.5 plugin, call stack is below:
```
Run ./gradlew --parallel lintProdRelease
Starting a Gradle Daemon
Checking for connection to remote build cache...
Found connection to build cache server. Enable remote build cache
Type-safe project accessors is an incubating feature.
> Task :build-logic:convention:generateExternalPluginSpecBuilders FROM-CACHE
> Task :build-logic:convention:extractPrecompiledScriptPluginPlugins FROM-CACHE
> Task :build-logic:convention:compilePluginsBlocks FROM-CACHE
> Task :build-logic:convention:generatePrecompiledScriptPluginAccessors FROM-CACHE
> Task :build-logic:convention:generateScriptPluginAdapters FROM-CACHE
> Task :build-logic:convention:compileKotlin FROM-CACHE
> Task :build-logic:convention:compileJava NO-SOURCE
> Task :build-logic:convention:pluginDescriptors
> Task :build-logic:convention:processResources
> Task :build-logic:convention:classes
> Task :build-logic:convention:jar
> Task :build-logic:convention:inspectClassesForKotlinIC

FAILURE: Build failed with an exception.

actionable tasks: 4 executed, 6 from cache
* What went wrong:
Could not determine the dependencies of task ':app:lintReportProdRelease'.
> Could not create task ':app:injectCrashlyticsBuildIdsProdRelease'.
   > Cannot invoke method map() on null object
   ```

**Comment by mrober** [comment]
We are aware of this issue. The work around is to continue to use version 2.9.2 for now. We are doing some internal refactoring to make the Crashlytics Gradle plugin more stable. I will post an artifact here when it is ready, so you don't have to wait for another release.

**Comment by mrober** [comment]
Hi folks, we build a preview fix for this issue. Feel free to try this out now instead of waiting for the next release. You can download and unzip the [m2repository.zip](https://github.com/firebase/firebase-android-sdk/files/11355433/m2repository.zip) and then add it to your build by doing something like:

```
pluginManagement {
  repositories {
    maven { url "/path/to/artifacts" }
    ...
  }
}
```

And then use the plugin by

`id 'com.google.firebase.crashlytics' version '2.9.6-alpha' apply false`

Please reply here if you have any more issues.

**Comment by runmael** [comment]
With `2.9.6-alpha` compiling finishes successfully 👍

**Comment by rmmmmm41** [comment]
Hi,@mrober,Same issue by use v2.9.6-alpha.The issue seems not be related to android studio version.I have tried to do it with android studio v2022.2.1 and v2022.1.1.

AGP & Gradle plugin versions:
7.2.2&7.5.1

**Comment by irfanadil** [comment]
I am having the same issue. That is why people are leaving android development.

**Comment by mrober** [comment]
Hi folks, by same issue do you mean the original issue "Cannot invoke method map() on null object" or the NoClassDefFoundError issue mentioned a few comments up?

I tried to reproduce the NoClassDefFoundError but cannot, even on a fresh app using all Android Studio defaults. The build tools is a fat jar that contains the httpclient, and I confirmed it is present in the artifact I posted. The only thing I can think of is maybe there is a conflict between it and other dependencies in your app? Would you be willing to send all your gradle build files to [Firebase support](https://firebase.google.com/support/) and mention me mrober, so I can take a deeper look?

**Comment by Zayankovsky** [comment]
With `2.9.6-alpha` I still get `Cannot invoke method map() on null object` error.

**Comment by rmmmmm41** [comment]
@mrober the original issue "Cannot invoke method map() on null object"

**Comment by mrober** [comment]
Thanks for the stacktrace @Zayankovsky, the `InjectBuildIdsTask.groovy:73` frame lines up with 2.9.5, not 2.9.6-alpha. Can you please double check your build points to 2.9.6-alpha? There should be a line like `id 'com.google.firebase.crashlytics' version '2.9.6-alpha' apply false` or `classpath 'com.google.firebase:firebase-crashlytics-gradle:2.9.6-alpha'` in your project level `build.gradle` file depending how your project is setup. If you have both, try removing one.

The disassembled bytecode you showed is a bit hard to read, so I am not sure what that does.

**Comment by Zayankovsky** [comment]
@mrober I use `classpath(com.google.firebase:firebase-crashlytics-gradle:2.9.6-alpha)`. I simply replaced `2.9.5` with `2.9.6-alpha` after extracting contents of `m2repository.zip` into my local maven folder and added `mavenLocal` to my list of repositories. I'm sure the build uses the library from `mavenLocal`, because without it the build fails with `Could not find com.google.firebase:firebase-crashlytics-gradle:2.9.6-alpha` error. 
Can you double check that the build you provided includes the fix? Android Studio shows the same decompiled code for `InjectBuildIdsTask` with `2.9.5` and `2.9.6-alpha`.

**Comment by mrober** [comment]
@Zayankovsky that was it, thanks! Here is the proper artifact [m2repository.zip](https://github.com/firebase/firebase-android-sdk/files/11502928/m2repository.zip). Sorry about that, I must have uploaded an earlier version of the artifact before. I double checked this one has the fix, please confirm this resolves the issue for you. Unfortunately this won't go out in the next release because I hadn't verified it in time, so it will be included in the next-next release.

**Comment by rmmmmm41** [comment]
Thanks，it works for me.

**Comment by Zayankovsky** [comment]
Works for me as well.

**Comment by mrober** [comment]
Thanks everyone! This fix will go out in the next-next release in a few weeks. If any other problems pop up with this artifact feel free to reopen this issue.

**Comment by egorikftp** [comment]
Hello, any ota about next release?

**Comment by mrober** [comment]
This is released now in 2.9.6.
